frontend/src/hooks/useDepartureTimeParamOrCurrentTime.ts

```diff
@@ -0,0 +1,66 @@
+import { formatISO, isValid, parseISO } from "date-fns";
+import { useEffect, useState } from "react";
+import { useStateParams } from "./useStateParams";
+
+/**
+ * Uses departure time set as URL param and checks if a valid date is given. If URL param is not present or an invalid
+ * date, current time gets used. Result is a formatted ISO date string.
+ */
+export function useDepartureTimeParamOrCurrentTime(): [string, (newDepartureTime: string) => void, () => void] {
+
+  const [departureTimeParam, setDepartureTimeParam] = useStateParams<string | null>(null, "departureTime", String, String);
+  const [departureTime, setDepartureTime] = useState<string>(getCurrentTimeISO());
+
+  /**
+   * When loading page, URL param departure times is checked to be present and a valid (ISO formatted) date string.
+   * If present+valid date, departureTime gets instantiated with this date string as well. If date is invalid, current
+   * time gets used to initialise both departureTime and URL param (overwrite invalid value). In the specific case of
+   * URL param not being present at all, URL param stays uninitialised.
+   */
+  useEffect(() => {
```

Comment:
Why this is the effect executed twice on page load (expected behaviour: one time? But seems to be "normal" to other effects/hooks, too.

Reply:
This is happing because we have strict mode enabled in dev mode, so we can catch bugs. This wont happen in prod builds.